source/astra-sdk/AuthManager.ts

```diff
@@ -7,8 +7,9 @@ import { inject, injectable } from 'inversify';
 import TYPES from '../../ioc.types';
 import { Logger } from 'winston';
 
-// tslint:disable-next-line:no-string-literal no-var-requires
-global['fetch'] = require('node-fetch');
+import fetch from 'fetch-with-proxy';
```

Comment:
If I read things right the typical pattern is usually tunneling over HTTP when going through a proxy regardless of the requested protocol (https)? Since fetch-with-proxy exposes the ability to point to a proxy that supports HTTPS we can get around some quirks in the cognito api that aren't compatible with the typical HTTP path that most libraries take?

Reply:
* amazon-cognito-identity-js package is semi-agnostic of the http request library. It uses the window fetch form, but requires a polyfill in old browsers and in nodeJS
* Normally, many http request libraries use the HTTP_PROXY and HTTPS_PROXY (etc) env variables to automatically proxy in explicit mode. Whether HTTPS is proxied over HTTP or HTTPS varies, with the former being more prevalent.
* node-fetch library (which was the polyfill in place for data ingestion agent) doesn't appear to support proxy configuration via ENV vars (https://github.com/bitinn/node-fetch/issues/195). ENV vars are pretty important for configuration, as some outbound requests are in library code that we don't own, so we can't do further configuration on them.

Reply:
Ah ok